ff_dir.c

```diff
@@ -135,9 +135,9 @@ static BaseType_t FF_ValidShortChar( char cChar );
 #endif /* if ( ffconfigLFN_SUPPORT != 0 ) */
 
 #if ( ffconfigUNICODE_UTF16_SUPPORT != 0 )
-    static void FF_MakeNameCompliant( FF_T_WCHAR * pcName );
+    static FF_Error_t FF_MakeNameCompliant( FF_T_WCHAR * pcName );
 #else
-    static void FF_MakeNameCompliant( char * pcName );
+    static FF_Error_t FF_MakeNameCompliant( char * pcName );
```

Comment:
> So please change it to :
> 
> ```
>     static BaseType_t FF_MakeNameCompliant( char * pcName );
> ```
Or even change the name:
~~~diff
-    static BaseType_t FF_MakeNameCompliant( char * pcName );
+    static BaseType_t FF_CheckNameCompliance( char * pcName );
~~~
because it will not change the name anymore.

But once again, thanks for your issue report #61 and thank you for the solution!

Reply:
What about the case of the first filename character being '\xE5'?
The character has to be remapped to 0x05 to have no collision with the file deleted marker (which is 0xE5).

My tests showed that, **filenames starting with '\xE5' still result in duplicated file-entries.**

To handle this case the function `FF_MakeNameCompliant` has to change the filename.